snakemake/cli.py

```diff
@@ -242,8 +242,17 @@ def parse_config(entries):
     """Parse config from args."""
     import yaml
 
-    yaml_base_load = lambda s: yaml.load(s, Loader=yaml.loader.BaseLoader)
-    parsers = [int, float, _bool_parser, yaml_base_load, str]
+    class SafeLoaderTimestamp(yaml.SafeLoader):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.add_constructor("tag:yaml.org,2002:timestamp", self.construct_yaml_str)
```

Comment:
I must say I am a bit unsure about this. Seems too implicit to me to automatically parse dates/timestamps. I think it is better if users do that themselves if they need it. Using the safe loader is good though.

Reply:
Mhm, this is indeed an interesting question. I think yes, this is the default behavior of yaml in python, hence it makes sense to me.

Reply:
Just to add my 2 cents I’d say the default safeloader behaviour is inconsistent because if you then write the config with datetime object back to YAML it’ll be different to the starting file.

@ollie-bell’s implementation overrides the default implicit behaviour (which is to parse date time strings as datetime objects) by replacing the datetime constructor with the str constructor. 

I think you could rename SafeLoaderTimestamp to something like SafeLoaderNoTimestamp to make this clear.